### ISSUE TITLE
The generate descriptor_set functionality does not handle relative paths properly

### ISSUE THREAD

**Issue by coryt** [uber/prototool#389]
`prototool version`
```                                                                                                                                        
Version:                 1.4.0-dev
Default protoc version:  3.7.0
Go version:              go1.12
OS/Arch:                 darwin/amd64
```
`prototool generate proto/activity --debug `
```
returning ProtoSets     {"workDirPath": "/Users/cory.t/go/src/github.com/xyz/service-home", "dirPath": "proto/activity", "protoSets": [{"WorkDirPath":"/Users/cory.t/go/src/github.com/xyz/service-home","DirPath":"/Users/cory.t/go/src/github.com/xyz/service-home/proto/activity","DirPathToFiles":{"/Users/cory.t/go/src/github.com/xyz/service-home/proto/activity":[{"Path":"/Users/cory.t/go/src/github.com/xyz/service-home/proto/activity/activity_api.proto","DisplayPath":"proto/activity/activity_api.proto"}]},"Config":{"DirPath":"/Users/cory.t/go/src/github.com/xyz/service-home/proto/activity","ExcludePrefixes":[],"Compile":{"ProtobufVersion":"3.6.1","IncludePaths":["/Users/cory.t/go/src"],"IncludeWellKnownTypes":true,"AllowUnusedImports":false},"Create":{"DirPathToBasePackage":null},"Lint":{"Group":"google","NoDefault":false,"IncludeIDs":[],"ExcludeIDs":[],"IgnoreIDToFilePaths":{},"FileHeader":"","JavaPackagePrefix":"","AllowSuppression":false},"Break":{"IncludeBeta":true,"AllowBetaDeps":true},"Gen":{"GoPluginOptions":{"ImportPath":"github.com/xyz/service-home/","ExtraModifiers":{"google/api/annotations.proto":"google.golang.org/genproto/googleapis/api/annotations","google/api/http.proto":"google.golang.org/genproto/googleapis/api/annotations"}},"Plugins":[{"Name":"descriptor_set","Type":1,"Flags":"","OutputPath":{"RelPath":"../../pkg/activity/api_descriptors/","AbsPath":"/Users/cory.t/go/src/github.com/xyz/service-home/pkg/activity/api_descriptors"},"FileSuffix":"pb","IncludeImports":true,"IncludeSourceInfo":true}]}}}]}
using file      {"file": "proto/activity/activity_api.proto"}
output from protoc --version    {"output": "libprotoc 3.6.1"}
protobuf already downloaded     {"path": "/Users/cory.t/Library/Caches/prototool/Darwin/x86_64/protobuf/3.6.1"}
running protoc  {"command": "/Users/cory.t/Library/Caches/prototool/Darwin/x86_64/protobuf/3.6.1/bin/protoc -I /Users/cory.t/go/src -I /Users/cory.t/Library/Caches/prototool/Darwin/x86_64/protobuf/3.6.1/include --descriptor_set_out=/Users/cory.t/go/src/github.com/xyz/service-home/pkg/activity/api_descriptors/..pb --include_imports --include_source_info /Users/cory.t/go/src/github.com/xyz/service-home/proto/activity/activity_api.proto"}
```

the file that gets written is `--descriptor_set_out=/Users/cory.t/go/src/github.com/xyz/service-home/pkg/activity/api_descriptors/..pb`

here is the generate prototool config:
```
generate:
  go_options:
    import_path: github.com/xyz/service-home/
    # Extra modifiers to include with Mfile=package.
    extra_modifiers:
        google/api/annotations.proto: google.golang.org/genproto/googleapis/api/annotations
        google/api/http.proto: google.golang.org/genproto/googleapis/api/annotations
  # The list of plugins.
  plugins:
    - name: go
      type: go
      flags: plugins=grpc
      output: ../../pkg/activity/gen/
    - name: descriptor_set
      output: ../../pkg/activity/api_descriptors/
      file_suffix: pb
      include_imports: true
      include_source_info: true
```

**Comment by peter-edge** [comment]
Can you try `output` without the trailing slash, so i.e. `../../pkg/activity/api_descriptors`? If this is the issue, I can fix internally.

**Comment by peter-edge** [comment]
@coryt I've been playing with this locally and can't replicate the issue. I do, however see where the issue is happening, but I can't explain it based on your above input. Can you do `go get github.com/uber/prototool/cmd/prototool@issue-389`, and then re-run the above, and give me the output? There should be `XXXX` lines, this is what I added:

```
$ git diff dev
diff --git a/internal/protoc/compiler.go b/internal/protoc/compiler.go
index e994fed..ebb8942 100644
--- a/internal/protoc/compiler.go
+++ b/internal/protoc/compiler.go
@@ -502,6 +502,10 @@ func getRelOutputFilePath(protoSet *file.ProtoSet, dirPath string, fileSuffix st
                // this should never happen, but could in a bad case with links
                return "", fmt.Errorf("could not find relative path for %q to %q, this is a system error, please file a bug at github.com/uber/prototool/issues/new: %v", dirPath, protoSet.Config.DirPath, err)
        }
+       fmt.Fprintf(os.Stderr, "XXXX configDirPath: %s dirPath: %s relPath: %s\n", protoSet.Config.DirPath, dirPath, relPath)
+       if relPath == "" || relPath == "." {
+               relPath = "out"
+       }
        return filepath.Join(relPath, filepath.Base(relPath)+"."+fileSuffix), nil
 }
```

**Comment by peter-edge** [comment]
Issue is here: https://github.com/uber/prototool/blob/2609cdd8ec2288707d9c182429057d1ac79cfa98/internal/protoc/compiler.go#L505

Tldr if `relPath` computes to the same as `protoSet.Config.DirPath`, then we get `""` for `relPath`, which computes to `"."` when passed to `filepath.Base`. This is a pretty bad oversight on my part and needs to be fixed and tested.

I can't understand why in the above that `dirPath == `protoSet.Config.DirPath`, this needs investigation as well.

**Comment by peter-edge** [comment]
This is fixed on dev.

**Comment by coryt** [comment]
thanks for the quick fix!
